### DISCUSSION TITLE
Subdomains

### DISCUSSION THREAD

**Discussion by ghost** [pocketbase/pocketbase#814] [Q&A] [answered]
Hi,
The project is amazing.

I would like to use pocketbase for a project with different subdomains: sub1.domain.com, sub2.domain.com, sub3.domain.com, etc.

Can pocketbase handle subdomains?
Or I have to create separate echo hosts like in this [example](https://echo.labstack.com/cookbook/subdomains/) and then use app.Dao() to manage the collections?

Thank you for your help.

**Comment by ganigeorgiev** [comment, marked as answer]
No, you can't have a single PocketBase instance binding to multiple domains/subdomains.

What you can do instead is setup a reverse proxy (eg. nginx) and start multiple PocketBase localhost instances on different ports and `pb_data` directories:

```go
./pocketbase serve --dir="./pb_data1" --http="127.0.0.1:8091" & \
./pocketbase serve --dir="./pb_data2" --http="127.0.0.1:8092" &

# to kill all instances:
sudo killall pocketbase
```

For an example nginx config, you check - https://pocketbase.io/docs/going-to-production/#deploy-behind-reverse-proxy

**Comment by benallfree** [comment]
@andrei-nita If you want separate instances of pocketbase running on separate domains, also check out https://github.com/benallfree/pockethost. That is a completely dockerized solution supporting subdomains. It is powering https://pockethost.io which allows multiple users to create their own subdomains. You wouldn't need that part, but you can maybe draw inspiration from it.

**Reply by pragmatech** [reply]
@benallfree you speak so blithely of the innards of pockethost.io (of which I am a happy 'customer') ...
and yet, stitching all the pkgs together, and the rest, sort of makes my eyes gloss over ...
Would you (or any others) be willing to give any pointers to one wanting to grok this project's assemblage?
(like in a zoomed out way for analysis by caveman, but persistent, devs ...)
thanks.
ps- just now, seeing the flux and update newness of all this, I reckon my above question might be a little lazy. (I still go with yarn to save the labor of typing 'npm run'. Typescript? Don't get me started ... This dev-thing is curious in that it promotes clever short-cuts ... and yet one need 'do the work'. I'll try and just study it too.)

**Reply by benallfree** [reply]
@pragmatech Head over to https://github.com/benallfree/pockethost/discussions/new and ask anything you want about pockethost, I try to keep these conversations focused on PocketBase since this forum is for that purpose. If there is anything in the readme that isn't clear, please do ask or if you solve it on your own you are always invited to create a PR to clarify the docs.
